cores/esp32/IPAddress.h

```diff
@@ -23,29 +23,27 @@
 #include <stdint.h>
 #include <WString.h>
 #include <Printable.h>
+#include <array>
 
 // A class to make it easier to handle and pass around IP addresses
 
 class IPAddress: public Printable
 {
 private:
-    union {
-        uint8_t bytes[4];  // IPv4 address
-        uint32_t dword;
-    } _address;
+	alignas(alignof(uint32_t)) std::array<uint8_t,4> _address{}; // IPv4 address
```

Comment:
Does `std::array` initialize with a zero value?
You did remove the default constructor, so it must be set to zero here.